discovery_engine_core/core/src/stack/ops/trusted.rs

```diff
@@ -89,10 +81,10 @@ impl Ops for TrustedNews {
         _key_phrases: &[KeyPhrase],
         history: &[HistoricDocument],
         stack: &[Document],
-    ) -> Result<Vec<Article>, GenericError> {
+    ) -> Result<Vec<Article>, NewItemsError> {
         let sources = Arc::new(self.sources.read().await.clone());
         if sources.is_empty() {
-            return Err(Box::new(Error::NoTrustedSources));
+            return Err(NewItemsError::NotReady);
```

Comment:
:grey_question: Why couldn't we just return `return Ok(vec![]);`

Reply:
this should indeed result in the same logical behavior in `update_stacks()` as this is basically what we did in the personalized news stack before. but this wouldn't allow us to differentiate why the articles are empty, either if they are actually empty or if `new_items()` failed for specific reasons, like missing key phrases. imo it's clearer to encode this in various errors and then decide if/how to succeed/fail based on the specific error, like continue on `NotReady`.

Reply:
We are not returning an empty list because that will count as a stack that correctly did a call but there was no data to return. If we return an empty array when we are not able to request new data we will count that as a stack that ended successfully and if, for example, there is no network we will not return an error at all from other stacks that failed.